### PR TITLE
Update `aws_secretsmanager_secret_version` label in Terraform deployment example

### LINKED ISSUE #197: Different resource block labels "scimsession" and "scimsession_1" necessary?
Currently main.tf in /aws-fargate-terraform contains the following resource block definitions. 

```terraform
resource "aws_secretsmanager_secret" "scimsession" {
  name_prefix             = local.name_prefix
  # Allow `terraform destroy` to delete secret (hint: save your scimsession file in 1Password)
  recovery_window_in_days = 0

  tags                    = local.tags
}

resource "aws_secretsmanager_secret_version" "scimsession_1" {
  secret_id     = aws_secretsmanager_secret.scimsession.id
  secret_string = base64encode(file("${path.module}/scimsession"))
}
```

Since there is no chance of a namespace collision (since they would be `aws_secretsmanager_secret.scimsession` and `aws_secretsmanager_secret_version.scimsession_1`, respectively, is it necesasry for the `secret_version` to be labelled `scimsession_1`? If we can revert to `scimsession` for both it would clean things up and avoid confusion. 
Thanks to @ag-adampike for the assist.

### PR DESCRIPTION
**Pull request by ag-adampike**
This wee PR updates the label of the `aws_secretsmanager_secret_version` resource block for clarity and simplicity.

Fixes #197.

Co-Authored-By: Scott Lougheed <scott@scottlougheed.com>
Co-Authored-By: Bryan Black <106387451+black-bryan@users.noreply.github.com>

### CODE REVIEW

**Comment by scottisloud** [comment]
FWIW I was able to deploy several bridges with the modified `main.tf` file without issue. Obviously proper testing and review required, but as an initial test and POC, this change does not impact deploying new bridges.

File: aws-ecsfargate-terraform/main.tf [hunk 1]
```diff
@@ -150,18 +150,18 @@ resource "aws_ecs_service" "op_scim_bridge" {
     security_groups  = [aws_security_group.service.id]
   }
 
-  tags             = local.tags
+  tags = local.tags
```
Comment (grellyd) [inline aws-ecsfargate-terraform/main.tf:153]: Thanks for all the whitespace fixes. This is much more readable and consistent!

File: aws-ecsfargate-terraform/main.tf [hunk 2]
```diff
@@ -296,3 +296,8 @@ resource "aws_route53_record" "op_scim_bridge" {
     evaluate_target_health = true
   }
 }
+
+moved {
+  from = aws_secretsmanager_secret_version.scimsession_1
+  to   = aws_secretsmanager_secret_version.scimsession
+}
```
Comment (grellyd) [inline aws-ecsfargate-terraform/main.tf:303]: Can you tell me more about this structure? What is this communicating to the terraform system? That both filenames are valid, and to rename one to the other?

**Review by grellyd — COMMENTED** [review]
Can you folks give me some more historical context on the name collisions for `scimsession_1`? I see the update and the issue, but why were we worried about collisions previously, and why is that no longer an issue?

Scott asked a good question, and I'd love an answer 😛 The answer is clearly 'no it is not necessary', but why?

File: aws-ecsfargate-terraform/main.tf (hunk 2 above)
Comment (ag-adampike) [reply aws-ecsfargate-terraform/main.tf:303]: It's a method for cleanly refactoring Terraform code that will help us improve our deployment example with less disruption for existing customers:
https://www.terraform.io/language/modules/develop/refactoring#moved-block-syntax

**Comment by ag-adampike** [comment]
> Can you folks give me some more historical context on the name collisions for `scimsession_1`? I see the update and the issue, but why were we worried about collisions previously, and why is that no longer an issue?

It's likely guesswork to assume exactly _why_ it was named with the `_1` suffix to begin with (it was authored before my time), but it feels superfluous, incoherent with the surrounding code in this module, a little confusing, and dissimilar to how AWS secrets are managed in other Terraform examples (in my admittedly cursory research).

This small refactor and associated `move` block provided a nice way to test the effects of such an update, and helps lay the groundwork for making a future version of this Terraform deployment example more composable and customizable. 😙👌

**Review by ag-adampike — COMMENTED** [review]
Added one more tweak to ensure support for the `move` block. :-)

File: aws-ecsfargate-terraform/main.tf (hunk 2 above)
Comment (grellyd) [reply aws-ecsfargate-terraform/main.tf:303]: Fantastic. Thank you.

File: aws-ecsfargate-terraform/main.tf [hunk 3]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 0.14.0"
+  required_version = ">= 1.1"
```
Comment (grellyd) [inline aws-ecsfargate-terraform/main.tf:2]: Guessing this is the minimum version required for terraform to support the moved syntax?

**Review by grellyd — APPROVED** [review]
LGTM! ✅

**Comment by grellyd** [comment]
> > Can you folks give me some more historical context on the name collisions for `scimsession_1`? I see the update and the issue, but why were we worried about collisions previously, and why is that no longer an issue?
> 
> It's likely guesswork to assume exactly _why_ it was named with the `_1` suffix to begin with (it was authored before my time), but it feels superfluous, inconsistent with the surrounding code in this module, a little confusing, and dissimilar to how AWS secrets are managed in other Terraform examples (in my admittedly cursory research).
> 
> This small refactor and associated `move` block provided a nice way to test the effects of such an update, and helps lay the groundwork for making a future version of this Terraform deployment example more composable and customizable. 😙👌

Just to triple check, you folks have tested the transition from the old filename style to the new? The `move` block is effective as expected? 

If so, I will just trust your testing. This does not seem a complicated change.

**Comment by ag-adampike** [comment]
@scottisloud and I have tested by migrating a bridge deployed from `master` to this branch and it behaves as desired and expected! 🤩

That said, I am more than happy for you or anyone on your team to take it for a test run to confirm the same, should you have the time and inclination. ☺️

**Comment by scottisloud** [comment]
> @scottisloud and I have tested by migrating a bridge deployed from `master` to this branch and it behaves as desired and expected! 🤩
> 
> That said, I am more than happy for you or anyone on your team to take it for a test run to confirm the same, should you have the time and inclination. ☺️

Can confirm. Test process was:
- Deploy scim bridge from Master branch. 
- re-apply the plan from the modified main.tf from this PR. 
- re-re-apply the modified plan to check for any misbehaviour of the ‘move’ block of the move had previously been performed. 

All three applications of the plan succeeded. This would suggest that new deployments using this revision will succeed, and existing scim bridges needing say, a version update, will also succeed.

File: aws-ecsfargate-terraform/main.tf (hunk 3 above)
Comment (ag-adampike) [reply aws-ecsfargate-terraform/main.tf:2]: Yep!

**Comment by grellyd** [comment]
I'm content. Have at it!

**Pull request merged**